### PR TITLE
Add "RenderGoalAsBox" Setting

### LINKED ISSUE #2171: render goal without motion
## Describe your suggestion
When you render a goal block (default in green) it does this weird thing where it moves parts of the outline up and down. This would disable that.

## Settings
renderGoalAsStaticCube? renderGoalWithoutMotion? idk

## Context
If I'm mining stone it just kinda sucks a lot real bad.

## Final checklist
- [x] I know how to properly use check boxes 
- [x] I have not used any OwO's or UwU's in this issue, as much as I wanted to.

### PR DESCRIPTION
**Pull request by thecakeisalie25**
Does what it says, if this setting is set to true it'll disable the animations on the goals. 

Closes #2171 

## Screenshot:
screenshot

### CODE REVIEW

**Review by ZacSharp — CHANGES_REQUESTED** [review]
I would give it a name related to animations (e.g. `renderGoalStatic`/`renderGoalAnimations`), because otherwise it suggests that it overrides `renderGoalXZBeacon`. Also the beacon beam should become static when animations are disabled.

**Comment by thecakeisalie25** [comment]
> 
> 
> I would give it a name related to animations (e.g. `renderGoalStatic`/`renderGoalAnimations`), because otherwise it suggests that it overrides `renderGoalXZBeacon`. Also the beacon beam should become static when animations are disabled.

How about renderGoalAnimated?

**Comment by ZacSharp** [comment]
sure

File: src/main/java/baritone/utils/PathRenderer.java [hunk 1]
```diff
@@ -207,8 +207,14 @@ public static void drawDankLitGoalBox(Entity player, Goal goal, float partialTic
         double minX, maxX;
         double minZ, maxZ;
         double minY, maxY;
-        double y1, y2;
-        double y = MathHelper.cos((float) (((float) ((System.nanoTime() / 100000L) % 20000L)) / 20000F * Math.PI * 2));
+        double y, y1, y2;
+        if (settings.renderGoalAsBox.value) {
+            // y = 1 causes rendering issues when the player is at the same y as the top of a block for some reason
+            y = .999F;
```
Comment (scorbett123) [inline src/main/java/baritone/utils/PathRenderer.java:213]:
```suggestion
            y = 0.999F;
```

**Review by scorbett123 — CHANGES_REQUESTED** [review]

File: src/api/java/baritone/api/Settings.java [hunk 2]
```diff
@@ -598,7 +598,7 @@
     /**
      * Render the goal as just a box instead of sick animated thingy
```
Comment (ZacSharp) [inline src/api/java/baritone/api/Settings.java:599]: You missed to update the javadoc

**Review by ZacSharp — APPROVED** [review]

File: src/main/java/baritone/utils/PathRenderer.java [hunk 3]
```diff
@@ -207,8 +207,14 @@ public static void drawDankLitGoalBox(Entity player, Goal goal, float partialTic
         double minX, maxX;
         double minZ, maxZ;
         double minY, maxY;
-        double y1, y2;
-        double y = MathHelper.cos((float) (((float) ((System.nanoTime() / 100000L) % 20000L)) / 20000F * Math.PI * 2));
+        double y, y1, y2;
+        if (!settings.renderGoalAnimated.value) {
+            // y = 1 causes rendering issues when the player is at the same y as the top of a block for some reason
+            y = 0.999F;
```
Comment (leijurv) [inline src/main/java/baritone/utils/PathRenderer.java:213]: don't we all hate floating point 😹

**Pull request merged**